src/fdc/MSXtar.cc

```diff
@@ -105,16 +107,14 @@ void MSXtar::parseBootSector(const MSXBootSector& boot)
 	chrootSector = rootDirStart;
 	dataStart = rootDirStart + nbRootDirSectors;
 
-	// Bypass max cluster check in sectorToCluster
-	maxCluster = std::numeric_limits<decltype(maxCluster)>::max();
-	maxCluster = sectorToCluster(boot.nrSectors);
-	maxCluster = std::min(maxCluster, BAD_FAT);
+	clusterCount = (boot.nrSectors - dataStart) / sectorsPerCluster;
+	clusterCount = std::min(clusterCount, FAT12_MAX_CLUSTER_COUNT);
```

Comment:
Detail, style issue. This is a more a comment on the existing code (prior to your change).
I don't like reassigning the same variable. In this case it's equally clear to write this as:
```
clusterCount = std::min((boot.nrSectors - dataStart) / sectorsPerCluster,
                        FAT12_MAX_CLUSTER_COUNT);
```

Reply:
I’ll do it like that, but just a side note;

I’m not a huge fan of indenting by spaces, because you often need to reindent when you e.g. rename stuff and such, and it’s also easy to forget and then it gets out of whack, or accidentally use tabs instead of spaces, etc. Same applies to lining up comments at the end of a line. So in my personal style I would write it like:

    clusterCount = std::min(
        (boot.nrSectors - dataStart) / sectorsPerCluster,
        FAT12_MAX_CLUSTER_COUNT
    );